src/RemoteAPI/class-analytics-posts-api.php

```diff
@@ -79,4 +78,17 @@ class Analytics_Posts_API extends Remote_API_Base {
 	public function get_posts_analytics( $api_params ) {
 		return $this->get_items( $api_params, true ); // @phpstan-ignore-line
 	}
+
+	/**
+	 * Sets the request's options for the remote API call.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @return array<string, mixed> The array of options.
+	 */
+	protected function set_request_options(): array {
+		return array(
+			'timeout' => 30, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
```

Comment:
As far as I know Parsely API have a timout of `10s` so how about we use it directly instead of making it configurable via base class?

Reply:
As Parse.ly API endpoints evolve or get added, they may have different timeouts and we may want to be explicit about the values we set. Although we could hardcode a max value for everything, I think that configurability is useful, since it allows us to also pass other things if the need ever comes up.

Reply:
I like the customizability and flexibility of the current solution. Having said that, no strong feelings if you just prefer us to hardcode a value.

Reply:
I like less code 😅

Reply:
In this case, since we'll be possibly querying content generation APIs that may take some time to respond, I'll add a large timeout.

Reply:
If the timeout is going to be different then let's keep it and better to have tests to make sure these are working as expected.

Reply:
OK, keeping this part of the code as is then.